openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -874,6 +887,18 @@ private static NodeList<ImportDeclarationNode> removeUnusedImports(ModulePartNod
         return imports;
     }
 
+    private static boolean traversAnnotationNode(Optional<MetadataNode> recField, boolean hasConstraint) {
+        MetadataNode metadata = recField.get();
+        for (AnnotationNode annotation : metadata.annotations()) {
+            String annotationRef = annotation.annotReference().toString();
+            if (annotationRef.contains(CONSTRAINT)) {
```

Comment:
cant we use `startsWith()` instead of `contains()`, to reduce false positives?
```suggestion
            if (annotationRef.startsWith(CONSTRAINT)) {
```

Reply:
yeah, we can add